module/VuFind/sql/migrations/pgsql/11.0/004-modify-table-structures-and-rename-indexes.sql

```diff
@@ -0,0 +1,131 @@
+
+-- Update external_session indexes
+DROP INDEX IF EXISTS external_session_id;
+CREATE INDEX IF NOT EXISTS external_session_id_idx ON external_session(external_session_id);
+CREATE UNIQUE INDEX IF NOT EXISTS external_session_session_id_idx ON external_session (session_id);
+
+-- Update oai_resumption index
+CREATE UNIQUE INDEX IF NOT EXISTS oai_resumption_token_idx ON oai_resumption (token);
+
+-- Update record table index
+CREATE UNIQUE INDEX IF NOT EXISTS record_record_id_source_index ON record (record_id, source);
+
+-- Add new auth_hash indexes
+CREATE INDEX IF NOT EXISTS auth_hash_session_id_idx ON auth_hash(session_id);
+CREATE UNIQUE INDEX IF NOT EXISTS auth_hash_hash_type_idx ON auth_hash(hash, type);
+
+-- Add new feedback indexes
+CREATE INDEX IF NOT EXISTS feedback_user_id_idx ON feedback (user_id);
+CREATE INDEX IF NOT EXISTS feedback_updated_by_idx ON feedback (updated_by);
+
+-- Add access_token index
+CREATE INDEX IF NOT EXISTS access_token_user_id_idx ON access_token (user_id);
+
+-- Remove old foreign key constraint from search table (if it exists)
+ALTER TABLE search DROP CONSTRAINT IF EXISTS search_ibfk_1;
+
+-- Remove old constraints from user_resource table
+ALTER TABLE user_resource DROP CONSTRAINT IF EXISTS user_resource_ibfk_1;
+ALTER TABLE user_resource DROP CONSTRAINT IF EXISTS user_resource_ibfk_2;
+ALTER TABLE user_resource DROP CONSTRAINT IF EXISTS user_resource_ibfk_3;
+ALTER TABLE user_resource DROP CONSTRAINT IF EXISTS user_resource_ibfk_4;
+ALTER TABLE user_resource DROP CONSTRAINT IF EXISTS user_resource_ibfk_5;
+
+-- Add new/updated foreign key constraints
+ALTER TABLE search ADD CONSTRAINT search_ibfk_1 FOREIGN KEY (user_id) REFERENCES "user" (id) ON DELETE CASCADE;
+
+ALTER TABLE user_card DROP CONSTRAINT IF EXISTS user_card_ibfk_1;
+ALTER TABLE user_card ADD CONSTRAINT user_card_ibfk_1 FOREIGN KEY (user_id) REFERENCES "user" (id) ON DELETE CASCADE;
+
+ALTER TABLE user_resource ADD CONSTRAINT user_resource_ibfk_1 FOREIGN KEY (user_id) REFERENCES "user" (id) ON DELETE CASCADE;
+ALTER TABLE user_resource ADD CONSTRAINT user_resource_ibfk_2 FOREIGN KEY (resource_id) REFERENCES resource (id) ON DELETE CASCADE;
+ALTER TABLE user_resource ADD CONSTRAINT user_resource_ibfk_5 FOREIGN KEY (list_id) REFERENCES user_list (id) ON DELETE CASCADE;
+
+ALTER TABLE login_token ADD CONSTRAINT login_token_ibfk_1 FOREIGN KEY (user_id) REFERENCES "user" (id) ON DELETE CASCADE;
+
+-- Clean up any duplicate or old index names
+DROP INDEX IF EXISTS last_used_idx;
+DROP INDEX IF EXISTS external_session_id;
+DROP INDEX IF EXISTS shortlinks_hash_idx;
+CREATE UNIQUE INDEX IF NOT EXISTS shortlinks_hash_idx ON shortlinks (hash);
+
+ALTER TABLE auth_hash DROP CONSTRAINT IF EXISTS auth_hash_hash_type_key;
+ALTER TABLE session DROP CONSTRAINT IF EXISTS session_session_id_key;
+ALTER TABLE external_session DROP CONSTRAINT IF EXISTS external_session_session_key;
```

Comment:
Is there a small typo here?

```suggestion
ALTER TABLE external_session DROP CONSTRAINT IF EXISTS external_session_session_id_key;
```

As-is, after running this migration, I'm still getting one failure from the test:

`DROP INDEX external_session_session_id_key`